doc/topics/blocking-vs-non-blocking.md

```diff
@@ -0,0 +1,140 @@
+# Overview of Blocking vs Non-Blocking
+
+This overview covers the difference between blocking and non-blocking calls in
```

Comment:
I'd prefer to highlight, blocking and non-blocking to `blocking` and `non-blocking` . That way it draws the readers eyes.